envconfig.go

```diff
@@ -319,19 +323,27 @@ func processField(value string, field reflect.Value) error {
 	case reflect.Map:
 		mp := reflect.MakeMap(typ)
 		if len(strings.TrimSpace(value)) != 0 {
-			pairs := strings.Split(value, ",")
+			kvDelimiter := ":"
+			if specifiedKvDelimiter, ok := tags.Lookup("kv_delimiter"); ok {
+				kvDelimiter = specifiedKvDelimiter
+			}
+			pairDelimiter := ","
+			if specifiedPairDelimiter, ok := tags.Lookup("delimiter"); ok {
+				pairDelimiter = specifiedPairDelimiter
+			}
+			pairs := strings.Split(value, pairDelimiter)
 			for _, pair := range pairs {
-				kvpair := strings.Split(pair, ":")
+				kvpair := strings.Split(pair, kvDelimiter)
```

Comment:
If you use SplitN instead of split even if the delimiter appears in the value it will not matter
```suggestion
				kvpair := strings.SplitN(pair, kvDelimiter, 2)
```

Reply:
thanks!